### PR TITLE
ESLint

### LINKED ISSUE #59: ESLint?
Should we use ESLint instead of JSHint? I see ESLint is already used, so should we swap out and do ember-cli-eslint to use all ESLint?

### PR DESCRIPTION
**Pull request by rwwagner90**
This will need to be merged after #58, as the changes for that are also here, since I just kept working on the same branch.

This just uses ESLint for everything rather than JSHint/JSCS.

Resolves #59

### CODE REVIEW

**Comment by rwjblue** [comment]
Needs a rebase (#58 landed), and seems to be failing CI.

**Comment by rwjblue** [comment]
Seems like most of these commits should be dropped out (all but the last one), right?

**Comment by rwwagner90** [comment]
@rwjblue yeah, they should be. The build will still be failing though. Can we drop node 0.10 from the tests, since Ember-CLI now bumped to >= 0.12?

**Comment by rwjblue** [comment]
Ya, dropping 0.10 seems fine now, but I don't think we can drop 0.12 yet...

**Comment by rwwagner90** [comment]
Yeah, still have some fixing to do

**Comment by rwwagner90** [comment]
@rwjblue got it fixed and dropped node 0.10

File: lib/coverage-instrumenter.js [hunk 1]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (rwjblue) [inline lib/coverage-instrumenter.js]: I'd like to keep these `'use strict';`'s

File: lib/config.js [hunk 2]
```diff
@@ -6,8 +6,6 @@
  * @property {Array<String>} reporters - list of reporters
  */
 
-'use strict';
```
Comment (rwjblue) [inline lib/config.js]: Please keep

File: lib/babel-istanbul-instrumenter.js [hunk 3]
```diff
@@ -3,8 +3,6 @@
  * @reference: https://github.com/douglasduteil/isparta/blob/master/src/instrumenter.js
  */
 
-'use strict';
```
Comment (rwjblue) [inline lib/babel-istanbul-instrumenter.js]: please keep

File: lib/attach-middleware.js [hunk 4]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (rwjblue) [inline lib/attach-middleware.js]: please keep

File: index.js [hunk 5]
```diff
@@ -1,5 +1,4 @@
 /* jshint node: true */
-'use strict';
```
Comment (rwjblue) [inline index.js]: please keep

File: config/environment.js [hunk 6]
```diff
@@ -1,5 +1,4 @@
 /*jshint node:true*/
-'use strict';
```
Comment (rwjblue) [inline config/environment.js]: please keep

File: config/release.js [hunk 7]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (rwjblue) [inline config/release.js]: please keep

File: ember-cli-build.js [hunk 8]
```diff
@@ -1,8 +1,6 @@
 /*jshint node:true*/
 /* global require, module */
 
-'use strict';
```
Comment (rwjblue) [inline ember-cli-build.js]: please keep

File: package.json [hunk 9]
```diff
@@ -9,7 +9,7 @@
   "scripts": {
     "node-test": "mocha test/**/*-test.js",
     "lint": "eslint config lib test tests *.js",
-    "jscs": "jscs lib test",
+    "jscs": "eslint lib test",
```
Comment (rwjblue) [inline package.json]: I think we can just remove this script (since there is no JSCS really)

File: package.json [hunk 10]
```diff
@@ -9,7 +9,7 @@
...
     "test": "npm run-script jscs && npm run-script lint && npm run-script node-test"
```
Comment (rwjblue) [inline package.json]: You can remove `npm run jscs` now

File: package.json [hunk 11]
```diff
@@ -18,7 +18,8 @@
   },
   "author": "Katie Gengler",
   "contributors": [
-    "Matthew Dahl (https://github.com/sandersky)"
+    "Matthew Dahl (https://github.com/sandersky)",
+    "Robert Wagner (https://github.com/rwwagner90)"
```
Comment (rwjblue) [inline package.json]: 👍 (I guess I should jump in here too 😛 )

File: package.json [hunk 12]
```diff
@@ -28,6 +29,7 @@
     "ember-cli": "2.5.0",
     "ember-cli-app-version": "^1.0.0",
     "ember-cli-dependency-checker": "^1.2.0",
+    "ember-cli-eslint": "2.0.0",
```
Comment (rwjblue) [inline package.json]: Lets use `^2.0.0`

File: test/helpers/run-command.js [hunk 13]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (rwjblue) [inline test/helpers/run-command.js]: please keep

File: test/integration/coverage-test.js [hunk 14]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (rwjblue) [inline test/integration/coverage-test.js]: please keep

File: test/unit/index-test.js [hunk 15]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (rwjblue) [inline test/unit/index-test.js]: please keep

File: testem.js [hunk 16]
```diff
@@ -1,7 +1,5 @@
 /*jshint node:true*/
 
-'use strict';
```
Comment (rwjblue) [inline testem.js]: please keep

File: testem.js [hunk 17]
```diff
@@ -1,7 +1,5 @@
 /*jshint node:true*/
```
Comment (rwjblue) [inline testem.js]: remove

**Comment by rwwagner90** [comment]
@rwjblue I think I got all the changes. PTAL

File: package.json [hunk 18]
```diff
@@ -9,16 +9,17 @@
...
   },
   "repository": "kategengler/ember-cli-code-coverage",
   "engines": {
     "node": ">= 0.10.0"
   },
   "author": "Katie Gengler",
   "contributors": [
-    "Matthew Dahl (https://github.com/sandersky)"
+    "Matthew Dahl (https://github.com/sandersky)",
+    "Robert Jackson (https://github.com/rwjblue)",
```
Comment (rwjblue) [inline package.json]: Thank you 😊

**Comment by rwjblue** [comment]
Looks like that change makes this need a rebase (it won't kick of a travis job if it can't merge cleanly).

**Comment by rwwagner90** [comment]
Looks like we're green now @rwjblue. Just waiting on your approval 😄

**Comment by rwjblue** [comment]
The big green button honors are all yours 😸

**Comment by rwwagner90** [comment]
Woo big green button! Thanks @rwjblue 🎉

**Pull request merged**